### ISSUE TITLE
Improve documentation

### ISSUE THREAD

**Issue by Lecrapouille** [the-moisrex/webpp#12]
Hi ! This project seems promising but the README should describe a little more the goal/achievement and what this library is offering and doing more smoothly than other existing projects ... In the same idea, there is a missing README file in the examples folder to explain what they are doing.

I think missing basic project description and good documentation may people not understanding the goal and leave the page and lose potential developers/helpers for the project.

**Comment by the-moisrex** [comment]
Hi; thanks for your interest in thia project.
Yes you're right about the lack of documentation and guides.

I added a few lines about the reasons why this kinda projects are necessary for C++ nowadays in the Readme file. I'll add some more later on.

Unfortunately, I'm not in a comfortable situation for coding and developing the project full time right now but I'm trying to get around that and I hope I can start developing this project sooner than later.

And I've been waiting for someone to come and help me develop this project for years now 😬.

**Comment by Lecrapouille** [comment]
> And I've been waiting for someone to come and help me develop this project for years now

Same for mostly of GitHub projects and in a general way, apart from the interest of the project itself where people using it for solving a particular problem of them will usually contribute, for other people, I think it's difficult for them to contribute or use if software architecture, comments, doc, tests, examples, tutorials are not given to help them entering in the code of somebody else. Coding becomes obvious (even for the main developer) when the general goal is well described to make people got the point and ideally code could be generated.

**Comment by the-moisrex** [comment]
> > And I've been waiting for someone to come and help me develop this project for years now
> 
> Same for mostly of GitHub projects and in a general way, apart from the interest of the project itself where people using it for solving a particular problem of them will usually contribute, for other people, I think it's difficult for them to contribute or use if software architecture, comments, doc, tests, examples, tutorials are not given to help them entering in the code of somebody else. Coding becomes obvious (even for the main developer) when the general goal is well described to make people got the point and ideally code could be generated.

Yes; the project being unknown is known to me, and I can't actually start sharing the coolness of this project when it's not production ready yet, and unfortunately no one will help unless they know about the coolness of the project. So it's a cycle!

God willing, someday, hopefully soon, that'll change!
